tasks/main-blivet.yml

```diff
@@ -31,17 +31,40 @@
     name: "{{ package_info.packages }}"
     state: present
 
-- name: manage the pools and volumes to match the specified state
-  blivet:
-    pools: "{{ storage_pools|default([]) }}"
-    volumes: "{{ storage_volumes|default([]) }}"
-    use_partitions: "{{ storage_use_partitions }}"
-    disklabel_type: "{{ storage_disklabel_type }}"
-    pool_defaults: "{{ storage_pool_defaults }}"
-    volume_defaults: "{{ storage_volume_defaults }}"
-    safe_mode: "{{ storage_safe_mode }}"
-    diskvolume_mkfs_option_map: "{{ __storage_blivet_diskvolume_mkfs_option_map|d(omit) }}"
-  register: blivet_output
+- service_facts:
+
+- set_fact:
+    # For an explanation of the to_json|from_json silliness, see
+    # https://github.com/ansible-collections/community.general/issues/320
+    storage_cryptsetup_services: "{{ ansible_facts.services|to_json|from_json|json_query('*.name')|json_query('[?starts_with(@, `\"systemd-cryptsetup@\"`)]') }}"
```

Comment:
Here is a way to do it without json_query:
```
    storage_cryptsetup_services: "{{
      ansible_facts.services | dict2items | selectattr('key', 'match', '^systemd-cryptsetup@') |
      map(attribute='value') | map(attribute='name') | list
    }}"
```
however, on fedora 33, this gives me a value like this: `['systemd-cryptsetup@luks\\x2d7fa23e85\\x2d4729\\x2d43b2\\x2dcc7a\\x2d9608d8cdb732.service']`
not sure if that is the same result of using json_query

Reply:
Someone came in a while back and switched all of our `selectattr` usage to `json_query` since `selectattr` doesn't work with Jinja versions < 2.8. You can see the comment, look at blame, etc. in `storage/tests/test-verify-volume-mount.yml`.

Reply:
> Someone came in a while back and switched all of our `selectattr` usage to `json_query` since `selectattr` doesn't work with Jinja versions < 2.8. You can see the comment, look at blame, etc. in `storage/tests/test-verify-volume-mount.yml`.

Mea culpa.  That was based on my incomplete understanding of the problems of `selectattr` and `json_query`.  The problem with jinja27 is not `selectattr` in general, it is that certain filters are not available - specifically, the `equalto`, `equal`, `==` filters to test if a value in an object is equal to some given value.

We have just completed converting all of the system roles to work with jinja27, and we figured out how to get around this limitation, by using the `match` filter instead of `equalto` or `==` - https://github.com/linux-system-roles/logging/commit/48d21efb71a36073a75914bdc3251dbb7fa63e77#diff-fe77ca1acb10f971c92988a8aed5bb5ae631443b1a20668629ccc846c608205aL11-L12

In the case of this PR, we have to use `match` anyway to look for services beginning with `systemd-cryptsetup@`.

In addition, I have verified the code above works with ansible 2.8/jinja2.7/centos7 controller host.

Reply:
We have CI tests for ansible 2.8/jinja2.7/centos7 controller host - the statuses like `*/ansible-2.8` - so you can know if you make a change that breaks in this case.

Reply:
However, all of that being said, if you prefer to keep the `json_query` implementation, that's fine with me.

Reply:
I'm happy with it as it is now if you are. No need to fiddle with it any more.